intents/intent_data_transaction_ext_test.go

```diff
@@ -8,13 +8,297 @@ import (
 
 	"github.com/0xsequence/ethkit/ethwallet"
 	"github.com/0xsequence/ethkit/go-ethereum/common"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/0xsequence/go-sequence"
 )
 
 func TestRecoverTransactionIntent(t *testing.T) {
+	data := `{
+		"version": "1",
+		"name": "sendTransaction",
+		"issued": 0,
+		"expires": 0,
+		"data": {
+		  "code": "sendTransaction",
+			"identifier": "test-identifier",
+			"wallet": "0xD67FC48b298B09Ed3D03403d930769C527186c4e",
+			"network": "10",
+			"transactions": [
+			{
+			  "type": "transaction",
+			  "to": "0x479F6a5b0C1728947318714963a583C56A78366A",
+			  "value": "39381",
+			  "data": "0x3251ba32"
+			},
+			{
+			  "type": "erc20send",
+			  "tokenAddress": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+			  "to": "0x7b1Bd3474D789e18e2E329E2c53F819B6E687b4A",
+			  "value": "1000"
+			},
+			{
+			  "type": "erc721send",
+			  "tokenAddress": "0xF87E31492Faf9A91B02Ee0dEAAd50d51d56D5d4d",
+			  "to": "0x17fFA2d95b58228e1ECb0C6Ac25A6EfD20BA08E4",
+			  "id": "7",
+			  "safe": true,
+			  "data": "0x112233"
+			},
+			{
+			  "type": "erc1155send",
+			  "tokenAddress": "0x631998e91476da5b870d741192fc5cbc55f5a52e",
+			  "to": "0x91E8aC543C5fEDf9F3Ef8b9dA1500dB84305681F",
+			  "vals": [
+				{
+				  "id": "2",
+				  "amount": "5"
+				},
+				{
+				  "id": "500",
+				  "amount": "1"
+				}
+			  ],
+			  "data": "0x223344"
+			},
+			{
+				"type": "contractCall",
+        "to": "0x140d72763D1ce39Ad4E2e73EC6e8FC53E5b73B64",
+				"data": {
+					"abi": "fillOrKillOrder(uint256 orderId, uint256 maxCost, address[] fees, bytes data)",
+					"args": [
+						"48774435471364917511246724398022004900255301025912680232738918790354204737320",
+						"1000000000000000000",
+						"[\"0x8541D65829f98f7D71A4655cCD7B2bB8494673bF\"]",
+						{
+							"abi": "notExpired(uint256,string)",
```

Comment:
the nested encoding is cool.. but what is very confusing, and maybe is actually completely invalid, is the "notExpired(uint256,string)" function name..? the encoder will actually include the "notExpired" name, is that intended..? or do we just want "(uint256,string)" for the encoding without including the method selector into the encoding..?